encodings/fsst/src/compute/compare.rs

```diff
@@ -16,33 +17,71 @@ impl CompareFn<FSSTArray> for FSSTEncoding {
         rhs: &Array,
         operator: Operator,
     ) -> VortexResult<Option<Array>> {
-        match (rhs.as_constant(), operator) {
-            (Some(constant), _) if constant.is_null() => {
+        match rhs.as_constant() {
+            Some(constant) if constant.is_null() => {
                 // All comparisons to null must return null
                 Ok(Some(
                     ConstantArray::new(Scalar::null(DType::Bool(Nullability::Nullable)), lhs.len())
                         .into_array(),
                 ))
             }
-            (Some(constant), Operator::Eq | Operator::NotEq) => compare_fsst_constant(
-                lhs,
-                &ConstantArray::new(constant, lhs.len()),
-                operator == Operator::Eq,
-            )
-            .map(Some),
+            Some(constant) => {
+                compare_fsst_constant(lhs, &ConstantArray::new(constant, lhs.len()), operator)
+            }
             // Otherwise, fall back to the default comparison behavior.
             _ => Ok(None),
         }
     }
 }
 
-/// Specialized compare function implementation used when performing equals or not equals against
-/// a constant.
+/// Specialized compare function implementation used when performing against a constant
 fn compare_fsst_constant(
     left: &FSSTArray,
     right: &ConstantArray,
-    equal: bool,
-) -> VortexResult<Array> {
+    operator: Operator,
+) -> VortexResult<Option<Array>> {
+    let rhs_is_empty = match right.dtype() {
+        DType::Utf8(_) => {
+            let v = right.scalar().as_utf8().value();
+            v.is_some_and(|v| v.is_empty())
+        }
+        DType::Binary(_) => {
+            let v = right.scalar().as_binary().value();
+            v.is_some_and(|v| v.is_empty())
+        }
+        _ => vortex_bail!(
+            "FSST array RHS can only be Utf8 or Binary, given {}",
+            right.dtype()
+        ),
+    };
+
+    if rhs_is_empty {
```

Comment:
I don't have a good sense of how expensive FSST decompression and string comparison is, but I could imagine for some distributions of data, for EQ, it's advantageous to:
1. Compare the lengths vector to the constant length.
2. Perform string comparison only on those strings which have matching length.

And analogously for NEQ. I suppose this feels like a separate PR / investigation though. Clickbench lacks queries that exercise this but TPC-H has a few string comparisons that would perhaps benefit.

Reply:
yep, we were talking about that too. hard to know

Reply:
I guess you want to length dist, estimator